### ISSUE TITLE
Gnome Desktop Icons on Ubuntu 20.04LTS (devel)

### ISSUE THREAD

**Issue by TS-CUBED** [regolith-linux/regolith-desktop#276]
Hi,

I have installed the current Regolith Linux on Ubuntu 20.04 (development). Everything seems to work fine, but for two issues:

One is that the GNOME desktop icons show, which creates a window which i3 then manages. This can easily be closed with Super-Ctrl-q, but none of the usual GNOME ways to disable them seem to work.  I tried GNOME tweaks and 

gsettings set org.gnome.desktop.background show-desktop-icons falsgsetting get shows it as false, but the icons show up on the next login (gsetting get still shows it as false, though).

**Comment by cjohnston1158** [comment]
I can confirm the gnome desktop icon window issue.

**Comment by kgilmer** [comment]
I believe the key you want is `org.gnome.gnome-flashback desktop`.  I found this by running `xprop` and clicking on the window that was showing icons.  This revealed that gnome-flashback provides that window.  Then, I checked all dconf settings exported by gnome-flashback with `gsettings list-recursively org.gnome.gnome-flashback`  I tried a few of them until `gsettings set org.gnome.gnome-flashback desktop false` caused the change I wanted.  

I will be experimenting more with this and integrating the fix into ppa:regolith-linux/unstable once I clearly understand what each setting does, and can verify backwards compatibility.  

Please add any other issues you're experiencing with Regolith on 20.04!

**Comment by jfrcom** [comment]
Thanks, setting `org.gnome.gnome-flashback desktop false` fixed it for me as well.

However in the file [i3-gnome-flashback](https://github.com/regolith-linux/regolith-gnome-flashback/blob/master/i3-gnome-flashback) I'm pretty sure that it should be `org.gnome.gnome-flashback root-background` instead of `org.gnome.gnome.flashback root-background`.

**Comment by gilvegliach** [comment]
Settings org.gnome.gnome-flashback desktop to false removes the icons but doesn't allow me to set a custom background any longer. Setting the value temporarily to true, changing the background, and setting it again to false works only till I log out. Then I get a gray background again (not the regolith's default).

```
sudo apt -s install regolith-desktop
Reading package lists... Done
Building dependency tree
Reading state information... Done
regolith-desktop is already the newest version (2.58-1focal1).
```

```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
```

**Comment by gilvegliach** [comment]
It actually works setting also ```gsettings set org.gnome.gnome.flashback root-background true```. The background after reboot is still set to the default focal fossa's background (maybe a bug in my setup at this point).

**Comment by nickgrim** [comment]
For anyone else here looking for a fix, the above all worked for me. I did the following:
```
$ gsettings set org.gnome.gnome-flashback desktop false
$ gsettings set org.gnome.gnome-flashback root-background true
```
and I no longer get a window-of-icons, I can change the background using the Settings app, and it persists across logout.

**Comment by MalakymR** [comment]
> For anyone else here looking for a fix, the above all worked for me. I did the following:
> 
> ```
> $ gsettings set org.gnome.gnome-flashback desktop false
> $ gsettings set org.gnome.gnome-flashback root-background true
> ```
> 
> and I no longer get a window-of-icons, I can change the background using the Settings app, and it persists across logout.

Doesn't seem to be the case anymore? list-recursive doesn't show any .gnome-flashback either.

$ gsettings set org.gnome.gnome-flashback desktop false
No such schema “org.gnome.gnome-flashback”

**Comment by kgilmer** [comment]
It's working for me @MalakymR .  Are you certain you have gnome-flashback installed?

```bash
$ gsettings list-recursively org.gnome.gnome-flashback
org.gnome.gnome-flashback screensaver true
org.gnome.gnome-flashback idle-monitor true
org.gnome.gnome-flashback input-sources true
org.gnome.gnome-flashback notifications false
org.gnome.gnome-flashback input-settings true
org.gnome.gnome-flashback shell true
org.gnome.gnome-flashback screenshot true
org.gnome.gnome-flashback audio-device-selection true
org.gnome.gnome-flashback a11y-keyboard true
org.gnome.gnome-flashback desktop false
org.gnome.gnome-flashback root-background true
org.gnome.gnome-flashback end-session-dialog true
org.gnome.gnome-flashback clipboard true
org.gnome.gnome-flashback screencast true
org.gnome.gnome-flashback status-notifier-watcher false
org.gnome.gnome-flashback automount-manager true
org.gnome.gnome-flashback polkit true
org.gnome.gnome-flashback.desktop show-icons true
org.gnome.gnome-flashback.desktop draw-background true
org.gnome.gnome-flashback.desktop.background fade true
org.gnome.gnome-flashback.desktop.icons icon-size '48px'
org.gnome.gnome-flashback.desktop.icons column-spacing uint32 10
org.gnome.gnome-flashback.desktop.icons show-home true
org.gnome.gnome-flashback.desktop.icons sort-by 'name'
org.gnome.gnome-flashback.desktop.icons show-trash true
org.gnome.gnome-flashback.desktop.icons placement 'auto-arrange-icons'
org.gnome.gnome-flashback.desktop.icons row-spacing uint32 2
org.gnome.gnome-flashback.desktop.icons extra-text-width uint32 48
```

```bash
$ apt list --installed | grep gnome-flashback

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

gnome-flashback-common/focal-updates,focal-updates,now 3.36.3-0ubuntu1 all [installed,automatic]
gnome-flashback/focal-updates,now 3.36.3-0ubuntu1 amd64 [installed]
regolith-gnome-flashback/focal,now 2.6.2-1 amd64 [installed,automatic]
```

**Comment by MalakymR** [comment]
My bad, wasn't aware it was as separate package.

Though, I have managed to remove the icons without this. The Tweaks gui for lots of things don't seem to work anymore.

If you open gnome-shell-extension-prefs instead - "Desktop Icons" is one of the "Built-In" plugins and works immediately.

The show-home preference only toggles the "Show the personal folder on the desktop" in this preferences window. You can also use show-trash but the only setting left is size for desktop icons.

org.gnome.desktop.background show-desktop-icons still does nothing as mentioned at the start.

Comparing outputs whilst toggling the extension narrows down to this...
$ gsettings get org.gnome.shell disabled-extensions
@as []
$ gsettings get org.gnome.shell disabled-extensions
['desktop-icons@csoriano']

So this works for me (this would replace other disabled-extensions so need to check first...)
$ gsettings set org.gnome.shell disabled-extensions [\'desktop-icons@csoriano\']

**Comment by kgilmer** [comment]
Hmm, I'm not exactly sure but it seems like you're running gnome she'll rather than gone flashback. If you are fine with this than great, but just FYI it isn't Regolith.

**Comment by argenos** [comment]
I'm wondering if this is also available in 18.04? I'm not using Regolith, only gnome-flashback and only see `desktop-background` listed instead of `desktop` and no `root-background` either

**Comment by kgilmer** [comment]
@argenos hello :smile: .  No, gnome-flashback in 18.04 had a different gsettings schema.
